### DISCUSSION TITLE
Enquiry on Easy-Ship orders

### DISCUSSION THREAD

**Discussion by lowzijian** [amzn/selling-partner-api-models#3599] [General] [answered]
Current marketplace region: Singapore

I am doing an easy-ship integration with a private application, doing bulk schedule & other fulfillment process (such as picking ,packing ) before dispatching to the amazon courier. There are few questions that I would like to ask:
- how to identify it is an easy-ship order ? I had tried my luck to see if it is use the `EasyShipShipmentStatus` from the response of `GET /orders/v0/orders` as identifier.  Despite the fact there is "PendingSchedule" value  , this field will not appear until the order was scheduled pickup. It seems the values provided in doc for this particular field are not accurate at all, will this be fixed in the future
-  Is there a sample of the `printableDocumentsUrl` from the response of `POST /easyShip/2022-03-23/packages/bulk` ? Like how the url looks like ? What is its folder structure and naming of the files ? Is all printable documents (shipping labels ) of every order squeeze into a single pdf , or separately

**Comment by abhaysharma0901** [comment]
Currently, I only have sellers who use Easy Ship, so it's not an issue today. However, this may create a problem in the future. Were you able to find the solution you were looking for? How did you automate the order processing? 

I have an auto-processing code where the API calls are made synchronously, and at the end, I get the shipping label for Easy Ship orders. However, this process takes around 2-3 minutes to process one order due to multiple API calls being made to retrieve the label. 

Do you have any solution to process this order and get the label, let's say, in just 5 seconds?

**Comment by petetak** [comment]
@lowzijian did you ever resolve these issues?

**Comment by chrisskhoury** [comment, marked as answer]
Hello @lowzijian

For Easy Ship, We have the External Fulfillment APIs that are the go to place for managing all EasyShip, SellerFlex programs and more. From fetching orders to shipping.

Call the [getShipments](https://developer-docs.amazon.com/sp-api/reference/getshipments-1) operation to get the list of ES orders. You can also determine the program used to transport a shipment using the channelName and shippingType values that getShipments and getShipment return.

For the label, You will get a presigned URL to fetch the ship label file using [generateShipLabels](https://developer-docs.amazon.com/sp-api/reference/generateshiplabels) request. To determine file type, use the contentType header of the file response. The file retrieved will be in one of the following supported formats:

ZPL: x.application/zpl
PDF: application/pdf
PNG: image/png
Plain Text: text/plain
Hope this helps!

Best,
Chris
